store/tikv/mocktikv/cop_handler_dag.go

```diff
@@ -95,7 +95,11 @@ func (h *rpcHandler) handleCopDAGRequest(req *coprocessor.Request) *coprocessor.
 		chunks = appendRow(chunks, data, rowCnt)
 		rowCnt++
 	}
-	return buildResp(chunks, err)
+	counts := make([]int64, len(dagReq.Executors))
+	for offset := len(dagReq.Executors) - 1; e != nil; e, offset = e.GetSrcExec(), offset-1 {
+		counts[offset] = e.Count() - 1
```

Comment:
Why `- 1` here?

Reply:
Because we always call the `Next` extra time for a `nil` result, so we `-1` here. I will add a comment for this.